src/core/qgsmaptopixelgeometrysimplifier.cpp

```diff
@@ -202,11 +218,23 @@ std::unique_ptr< QgsAbstractGeometry > QgsMapToPixelSimplifier::simplifyGeometry
           {
             x = *xData++;
             y = *yData++;
+
+            if ( geometry.is3D() )
```

Comment:
should be a tiny bit faster:
```suggestion
            if ( zData )
```

Reply:
Fixed with bool variable